src/v/kafka/server/handlers/fetch.cc

```diff
@@ -281,6 +282,74 @@ make_partition_response_error(model::partition_id p_id, error_code error) {
     };
 }
 
+uint32_t
+control_record_size(int64_t ts_delta, int32_t offset_delta, const iobuf& key) {
```

Comment:
can we invoke `uint32_t record_batch_builder::record_size` from the record batch builder instead?

Reply:
In the `record_batch_builder` record_size is a private method that uses internal builder type. Additionally this is subset of the method content tailored to calculate control batch size.